src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1432,7 +1433,8 @@ private int appendClassName(char** ppBuf, ref int pnBufLen, CORINFO_CLASS_STRUCT
 
         private bool isValueClass(CORINFO_CLASS_STRUCT_* cls)
         {
-            return HandleToObject(cls).IsValueType;
+            TypeDesc type = HandleToObject(cls);
+            return type.IsValueType || type.IsPointer || type.IsFunctionPointer;
```

Comment:
It would be useful to update documentation on isValueClass in corinfo.h to describe this quirk.

@dotnet/jit-contrib `isValueClass` JIT/EE interface method returns true for pointers and function pointers. RyuJIT seems to depend on this non-obvious behavior in subtle ways. This change makes crossgen2 to match it. It would be trivial to fix the semantics of this method on EE side to only return true for true value types, but it would also require matching changes on the JIT side. Do you believe that it is worth opening a bug on this?

Reply:
I think it is worth an issue, would be interesting to see what changes when we fix it.

Reply:
> It would be trivial to fix the semantics of this method on EE side to only return true for true value types

Do you happen to have EE side fix that you can share for me to try?

Reply:
Something like this: https://github.com/jkotas/runtime/commit/35f590f6c9736438ece3558beaacb1141ce78d57

Reply:
Opened issue #44730